train_redpajama.py

```diff
@@ -226,14 +227,16 @@ def create_dataloader(
     batch_size: int,
     block_size: int,
     data_dir: str,
+    fabric,
     shuffle: bool = True,
     seed: int = 12345,
 ) -> DataLoader:
     datasets = []
     for prefix, _ in data_config:
         filenames = glob.glob(os.path.join(data_dir, prefix + "*"))
         dataset = PackedDataset(
-            filenames, n_chunks=10, block_size=block_size, shuffle=shuffle, seed=seed
+            filenames, n_chunks=4, block_size=block_size, shuffle=shuffle, seed=seed,
+            num_processes=fabric.world_size, process_rank=fabric.global_rank,
```

Comment:
the n_chunks value here must be smaller than the number of files ending up in each shard. It's unclear to me what exactly the chunks refer to

Reply:
Say you have 12 files (12 chunks, each chunk containing N blocks of tokens) and you set `n_chunks=4`.
The dataset will pre-load 4 chunks in memory and iterate over the blocks contained in those chunks, optionally shuffling the blocks. If the dataset doesn't find enough chunks to load, it will raise a StopIteration.

I think we should actually give the opportunity to rewind as well for overfit tests etc.